app/controllers/topbar_sweeper.rb

```diff
@@ -13,15 +13,15 @@ def after_destroy(record)
     record.expire_topbar_cache(self)
   end
 
-  def after_select(record)
-    record.expire_topbar_cache(self)
-  end
-
```

Comment:
It is a valid hook, it's supposed to trigger after you select a taxonomy (https://github.com/theforeman/foreman/blob/develop/app/controllers/concerns/foreman/controller/taxonomies_controller.rb#L99)

I've just tried and if you remove it, the taxonomies selector in the top right corner doesn't changewhen you change the taxonomy, unless you remove the fragment manually. (in production)